ExpSettingsVal.py

```diff
@@ -0,0 +1,331 @@
+# 	2 The assigned AWG must exist in the library
+#	3 The name of the PhysicalChannel channel must be of the form AWGName-AWGChannel
+#   4 Device specific naming conventions
+#     APS: 12, 34, 1m1, 2m1, 3m1, 4m1
+#     APS2: 12, 12m1, 12m2, 12m3, 12m4
+#     Tek5014: 12, 34, 1m1, 1m2, 2m1, 2m2, 3m1, 3m2, 4m1, 4m2
+
+# Conventions to be added
+# 
+#
+
+#####################################################################################
+## Helper functions for list comprehension 
+
+def is_logicalmarker_channel(name):
+	return is_channel_type(name, QGL.Channels.LogicalMarkerChannel)	
+
+def is_physicalmarker_channel(name):
+	return is_channel_type(name, QGL.Channels.PhysicalMarkerChannel)	
+
+def is_physicalIQ_channel(name):
+	return is_channel_type(name, QGL.Channels.PhysicalQuadratureChannel)	
+
+def is_qubit_channel(name):
+	return is_channel_type(name, QGL.Channels.Qubit)	
+
+def is_measurement_channel(name):
+	return is_channel_type(name, QGL.Channels.Measurement)	
+
+def requires_physical_channel(name):
+	return (is_channel_type(name, QGL.Channels.Qubit) or
+	        is_channel_type(name, QGL.Channels.LogicalMarkerChannel) or
+	        is_channel_type(name, QGL.Channels.Measurement))
+
+def is_channel_type(name, channelType):
+	channels = Libraries.channelLib.channelDict
+	return isinstance(channels[name], channelType)	
+
+#####################################################################################
+### Apply global rules
+
+def test_require_physical():
+	"""Enforces rule requiring physical channels for certain logical channels
+
+	   See requires_physical_channel() for list of Channel types requiring a
+	   Physical channel.
+	"""
+	errors = []
+	channels = Libraries.channelLib.channelDict
+	testChannels = [channelName for channelName in channels.keys() if requires_physical_channel(channelName)]
+
+	for channel in testChannels:
+		physicalChannel = channels[channel].physChan
+
+		if physicalChannel is None:
+			print '"{0}" channel "{1}" Physical Channel is not defined'.format(channels[channel].__class__.__name__, channel)
+			errors.append(channel)
+
+	return errors
+
+## Apply invidual test based on channel type
+def test_logical_channels():
+	"""
+		Enforces rules applied against logical channels
+
+		These are rules in addition to those applied at the global level.
+
+		Rules:
+		PhysicalChannel but be in library
+		LogicalMarkerChannel must map to PhysicalMarkerChannel
+	"""
+	errors = []
+	channels = Libraries.channelLib.channelDict
+	logicalChannels = [channelName for channelName in channels.keys() if is_logicalmarker_channel(channelName)]
+	
+	for channel in logicalChannels:
+		if not is_logicalmarker_channel(channel):
+			continue
+		errorHeader = 'LogicalMarkerChannel "{0}" requires a Physical Marker Channel'.format(channel)
+		if channels[channel].physChan is not None:
+			physicalChannelName = channels[channel].physChan.label
+			if physicalChannelName not in channels.keys():
```

Comment:
The intent was to allow for independence between tests, that is allow a test to fail but keep processing looking for other errors without running into key errors. I've moved things around a bit to simplify and make things cleaner.